webapp/src/test/java/org/apache/falcon/resource/ExtensionManagerIT.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.falcon.resource;
+
+import org.apache.falcon.extensions.ExtensionProperties;
+import org.apache.falcon.extensions.mirroring.hdfs.HdfsMirroringExtensionProperties;
+import org.apache.falcon.extensions.store.AbstractTestExtensionStore;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Unit tests for org.apache.falcon.extensions.ExtensionManager.
+ */
+public class ExtensionManagerIT extends AbstractTestExtensionStore {
+    private static final String HDFS_MIRRORING_PROPERTY_TEMPLATE = "/hdfs-mirroring-property-template.txt";
+    private static final String JOB_NAME_1 = "hdfs-mirroring-job-1";
+    private static final String JOB_NAME_2 = "hdfs-mirroring-job-2";
+    private static final String CLUSTER_NAME = "primaryCluster";
+    private static final String START_TIME_1 = "2016-05-03T00:00Z";
+    private static final String START_TIME_2 = "2016-05-01T00:00Z";
+    private static final String FREQUENCY = "days(1)";
+    private static final String SOURCE_DIR = "/apps/falcon/demo/input-{year}-{month}-{day}";
+    private static final String TARGET_DIR = "/apps/falcon/demo/output-{year}-{month}-{day}";
+
+    private final TestContext context = new TestContext();
+
+    @BeforeClass
+    public void prepare() throws Exception {
+        TestContext.prepare();
+        initExtensionStore(TestContext.class);
+    }
+
+    @AfterClass
+    public void tearDown() throws Exception {
+        TestContext.deleteEntitiesFromStore();
```

Comment:
You might also want to call cleanup() to delete extensionStorePath during tearDown()